### PR TITLE
Some updates to scream scripts

### PR DESCRIPTION
**Pull request by jgfouca**
Main change: Fully encapsulte env considerations in get_mach_env_setup_command  …

Some settings like OMP_PROC_BIND and CTEST_PARALLEL_LEVEL were not
being handled by this function. Now that they are, this increases the
usefullness of scream-env-cmd and improves the overall coherence of
tools.

Other changes:
* scripts-tests baseline normalization so that changes in HEAD's name do not cause diffs
* some cleanup of repeated code in machines_specs

### CODE REVIEW

File: components/scream/scripts/machines_specs.py [hunk 1]
```diff
@@ -42,8 +42,8 @@
     "mappy"   : (["module purge", "module load sems-env sems-python/3.5.2 sems-gcc/9.2.0 sems-cmake/3.12.2 sems-git/2.10.1 sems-openmpi/4.0.2 sems-netcdf"],
                  ["mpicxx","mpifort","mpicc"],
                   "",
-                  48,
-                  48,
+                  46,
```
Comment (bartgol) [inline components/scream/scripts/machines_specs.py:45]: You're a polite person. :)

File: components/scream/scripts/machines_specs.py [hunk 2]
```diff
@@ -106,97 +106,83 @@ def is_machine_supported(machine):
...
+    return mach_custom_env
 
 ###############################################################################
 def get_mach_cxx_compiler(machine):
 ###############################################################################
-    expect(is_machine_supported(machine),
-           "Machine {} is not currently supported by scream testing system.\n"
-           " Note: you can also create a file `~/.cime/scream_mach_specs.py` with your local machine specs.".format(machine))
-
+    assert_machine_supported(machine)
```
Comment (bartgol) [inline components/scream/scripts/machines_specs.py:131]: Nice reduction of code duplication.

File: components/scream/scripts/scripts-tests [hunk 3]
```diff
@@ -62,13 +62,17 @@ def run_cmd_store_output(test_obj, cmd, output_file):
 ###############################################################################
     output_file.parent.mkdir(parents=True, exist_ok=True)
     output = run_cmd_assert_result(test_obj, cmd, from_dir=TEST_DIR)
+    head = get_current_head()
+    output = output.replace(head, "CURRENT_HEAD_NORMALIZED")
```
Comment (bartgol) [inline components/scream/scripts/scripts-tests:66]: Curious: what is the reason for this?

**Review by bartgol — APPROVED** [review]

File: components/scream/scripts/scripts-tests (hunk 3 above)
Comment (bartgol) [reply components/scream/scripts/scripts-tests:66]: Interesting. I have always generated baselines with origin/master (I run with `-i -l` on my laptop), and never got diffs.

Comment (jgfouca) [reply components/scream/scripts/scripts-tests:66]: @bartgol , sounds like you're talking about test-all-scream. I'm talking about scripts-tests (which calls test-all-scream and stores all the terminal output as a baseline).

Comment (bartgol) [reply components/scream/scripts/scripts-tests:66]: Aaaaah, ok, sorry. Got confused. Thanks!

**Pull request merged**